[DebugInfo] Introduce DW_AT_LLVM_language_dialect (#194898)
llvm/llvm-project: llvm/docs/LangRef.rst

                        enums: !2, retainedTypes: !3, globals: !4, imports: !5,
                        macros: !6, dwoId: 0x0abcd)

The optional ``dialect:`` field encodes the source-language *dialect* of the
compile unit as an enum (the ``DW_LLVM_LANG_DIALECT_*`` family defined in

> ``include/llvm/BinaryFormat/Dwarf.def``). It corresponds to the

Reviewer: ```suggestion
compile unit as an enum. It corresponds to the
```

Reviewer: (since you enumerate the value below anyway)

``DW_AT_LLVM_language_dialect`` attribute emitted on ``DW_TAG_compile_unit`` and
is intended for language dialects that represent different execution models.
When specified, the field must name one of the known dialects (either
symbolically or with the matching numeric value):

* ``DW_LLVM_LANG_DIALECT_simt`` / ``1``
* ``DW_LLVM_LANG_DIALECT_tile`` / ``2``

Omitting the ``dialect:`` field is the only way to express "no dialect" in
textual IR; the assembly parser rejects an explicit ``dialect: 0`` because,
when present, the field must name a known dialect. The IR verifier and
bitcode reader treat a stored value of 0 as "no dialect specified" and
reject any value outside the defined enumeration. When set, the dialect is
printed in symbolic form; when unset, the field is omitted from the textual

> output and no ``DW_AT_LLVM_language_dialect`` attribute is emitted.

Reviewer: ```suggestion
```

Don't think we need to go into these details. Its not unique to the `dialect` field (i.e., all enum fields are treated this way in the textual representation)
